File: GXpert/GXpert.Web/Modules/ServerTypes/Skills/SkillCategoryColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SkillCategoryRow } from "./SkillCategoryRow";

export interface SkillCategoryColumns {
    Id: Column<SkillCategoryRow>;
    Name: Column<SkillCategoryRow>;
    Description: Column<SkillCategoryRow>;
}

export class SkillCategoryColumns extends ColumnsBase<SkillCategoryRow> {
    static readonly columnsKey = 'Skills.SkillCategory';
    static readonly Fields = fieldsProxy<SkillCategoryColumns>();
}

File: GXpert/GXpert.Web/Modules/Skills/SkillCategory/SkillCategoryColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Skills.Columns;

[ColumnsScript("Skills.SkillCategory")]
[BasedOnRow(typeof(SkillCategoryRow), CheckNames = true)]
public class SkillCategoryColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Description { get; set; }
  
}